data_structures/src/error.rs

```diff
@@ -200,12 +200,44 @@ pub enum TransactionError {
         expected: Vec<PublicKeyHash>,
         found: Vec<PublicKeyHash>,
     },
-    /// Invalid collateral
+    /// Invalid collateral in data request
     #[fail(
         display = "The specified collateral ({} nwits), is less than the minimum required ({} nwits)",
         value, min
     )]
     InvalidCollateral { value: u64, min: u64 },
+    /// Negative collateral in commit transaction
+    #[fail(
+        display = "Tried to create value in commit collateral. Input value: {}, output value: {}",
+        input_value, output_value
+    )]
+    NegativeCollateral { input_value: u64, output_value: u64 },
```

Comment:
Why dont use the IncorrectCollateral error for this case?

Reply:
Because the collateral value is negative, and the `IncorrectCollateral` error expects positive values.